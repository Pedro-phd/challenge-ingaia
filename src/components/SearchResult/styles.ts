import styled from 'styled-components'

export const Wrapper = styled.main`
  display: flex;
  max-width: 992px;
  flex-wrap: wrap;
  justify-content: center;
  margin: 0 auto;
`
