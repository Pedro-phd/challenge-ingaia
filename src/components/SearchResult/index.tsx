import * as S from './styles'
import { useState, useEffect } from 'react'
import { useQuery } from '@apollo/client'
import { GET_ALL_CHARACTERS } from '../../graphql/queries'
import { ICharacter, IInfoApi } from '../../types/index'
import Character from '../Character/index'
import Loading from 'components/Loading'

const SearchResult = () => {
  const { loading, error, data } = useQuery(GET_ALL_CHARACTERS, {
    variables: { page: 1 }
  })

  const [characters, setCharacters] = useState<Array<ICharacter>>([])
  const [info, setInfo] = useState<IInfoApi>()

  useEffect(() => {
    if (data != undefined) {
      setCharacters(data?.characters?.results)
      setInfo(data?.characters?.info)
    }
    console.log(data, error, info)
  }, [data, loading, error, info])

  return (
    <S.Wrapper>
      <Loading visible={loading} />
      {characters.length > 0 &&
        characters.map((character) => {
          return <Character key={character.id} {...character} />
        })}
    </S.Wrapper>
  )
}

export default SearchResult
