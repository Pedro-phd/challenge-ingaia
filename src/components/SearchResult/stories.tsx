import { Story, Meta } from '@storybook/react/types-6-0'
import SearchResult from '.'

export default {
  title: 'SearchResult',
  component: SearchResult
} as Meta

export const Default: Story = () => <SearchResult />
