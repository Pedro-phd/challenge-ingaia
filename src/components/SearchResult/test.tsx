import { render, screen } from '@testing-library/react'

import SearchResult from '.'

describe('<SearchResult />', () => {
  it('should render the heading', () => {
    const { container } = render(<SearchResult />)

    expect(
      screen.getByRole('heading', { name: /SearchResult/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
