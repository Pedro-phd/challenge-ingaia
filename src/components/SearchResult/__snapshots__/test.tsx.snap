// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    SearchResult
  </h1>
</main>
`
